recipes/quill/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans import ConanFile, CMake, tools
+from conans.model.version import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class QuillConan(ConanFile):
+    name = "quill"
+    description = "C++14 Asynchronous Low Latency Logging Library"
+    homepage = "https://github.com/odygrd/quill/"
+    topics = ("conan", "quill", "logging", "log")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fmt_external": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fmt_external": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```

Comment:
```suggestion
            del self.options.fPIC

    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "14")
        minimum_version = {
            "clang": 3.4,
            "gcc": 5,
            "Visual Studio": 14.0,
        }.get(str(self.settings.compiler))
        if not minimum_version:
            self.output.warn(
                "Unknown compiler: assuming compiler supports C++14")
        else:
            if Version(self.settings.compiler.version) < minimum_version:
                raise ConanInvalidConfiguration(
                    "The compiler {} {} does not support C++14".format(self.settings.compiler, self.settings.compiler.version))
```

C++14 is not supported by all compilers.

Reply:
Thanks a lot for reviewing, I will make the changes.

Also, only `x86-64` architecture is supported at the moment.

It won't compile for `arm`, `arm64`. 
Is there an equivalent `tools.check_arch(self, "x86-64")` check I could add ?

Reply:
Add it to configure(self):

```
if "arm" in str(self.settings.arch):
    raise ConanInvalidConfiguration("{} is not supported by {}".format(self.settings.arch, self.name))
```